### PR TITLE
feat: interview reviews base 🧱 

### PR DESCRIPTION
**Pull request by tomas-salgado**
## Description ✏️

This PR implements the first part of the interview reviews feature.
It adds

- `interview_reviews` table to the database
- Fetch and list all interview reviews for a given company
- Implementation of navigation between `Employees`, `Company Reviews`, and `Interview Reviews` on Companies page
- Put front-end changes in `_profile.companies_.$id.tsx` behind a new feature flag named `interview-reviews`

## Type of Change 🐞

- [X] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

**Comment by tomas-salgado** [comment]
I decided to break up the feature into multiple PRs-it was getting too messy and this makes more sense. I'm curious if I should create stacked PRs and then not merge any until all the PRs are approved, or if I should just put the interview reviews feature behind a feature flag, so that way we can merge them in one at a time but it won't appear to members until the entire feature is complete.

File: apps/member-profile/app/routes/_profile.companies_.$id.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import {
   type SerializeFrom,
 } from '@remix-run/node';
 import { generatePath, Link, useLoaderData } from '@remix-run/react';
+import { useSearchParams } from '@remix-run/react';
```
Comment (tomas-salgado) [inline apps/member-profile/app/routes/_profile.companies_.$id.tsx:7]: It seems there is also `useSearchParams` from `@oyster/ui` [here](https://github.com/tomas-salgado/oyster/blob/d34f0dc050bc819a52583a0338bc1148baf256be/packages/ui/src/hooks/use-search-params.ts). Noting to investigate if I should use this instead

File: apps/member-profile/app/routes/_profile.companies_.$id.tsx [hunk 2]
```diff
@@ -156,9 +184,24 @@ export default function CompanyPage() {
...
+          <NavigationItem to="?tab=interview-reviews">
+            Interview Reviews
+          </NavigationItem>
+        </ul>
+      </nav>
+      <Divider my="4" />
+      {activeTab === 'employees' && (
+        <>
+          <CurrentEmployees />
+          <PastEmployees />
```
Comment (tomas-salgado) [inline apps/member-profile/app/routes/_profile.companies_.$id.tsx:200]: Might make more sense to have `CurrentEmployees` and `PastEmployees` under a single component named `Employees`

**Comment by tomas-salgado** [comment]
One minor issue I was facing is that the navigation tabs here always show up green. This is different from the recaps page, for example, where the tab is only green if it is selected. Having trouble figuring out the issue here. 

Screenshot 2024-07-29 at 2 47 10 PM

**Comment by ramiAbdou** [comment]
Going to close for now, and we'll revisit this in the future!

**Pull request closed without merging**